packages/amplify-provider-awscloudformation/src/admin-modelgen.ts

```diff
@@ -80,6 +83,17 @@ export const adminModelgen = async (context: $TSContext, resources: $TSAny[]): P
   } finally {
     stateManager.setProjectConfig(undefined, originalProjectConfig);
     process.stdout.write = originalStdoutWrite;
+    if (tempStdoutWrite) {
+      await new Promise((resolve, reject) => {
+        tempStdoutWrite.close((err) => {
```

Comment:
there's no promise based version of writestream.close. But we have to wait for successful closure before removing directory.